packages/web/hooks/input/use-amount-input.ts

```diff
@@ -31,55 +39,62 @@ export function useAmountInput(currency?: Currency, inputDebounceMs = 500) {
   const rawCurrencyBalance = balances?.balances.find(
     (bal) => bal.denom === currency?.coinMinimalDenom
   )?.amount;
-
   // manage amounts, with ability to set fraction of the amount
   // `inputAmount` is the raw string input that includes decimals
   const [inputAmount, _setAmount] = useState("");
   const [fraction, setFraction] = useState<number | null>(null);
+
   const setAmount = useCallback(
     (amount: string) => {
       // check validity of raw input
       if (!isValidNumericalRawInput(amount)) return;
-
       if (amount.startsWith(".")) {
         amount = "0" + amount;
       }
 
       if (fraction != null) {
         setFraction(null);
       }
+
       _setAmount(amount);
     },
     [fraction]
   );
-
   // clear fraction when user changes currency
   // and user has no balance
   useEffect(() => {
     if (isBalancesFetched && !rawCurrencyBalance) setFraction(null);
   }, [isBalancesFetched, rawCurrencyBalance, currency]);
 
-  /** Amount derived from user input or from a fraction of the user's balance. */
+  const isHalf = useMemo(() => fraction === 0.5, [fraction]);
+  const isMax = useMemo(() => fraction === 1, [fraction]);
+
```

Comment:
I'm not completely sure if displaying the 'max' button as selected will effectively represent the current state, considering the maximum value can change. I think it might be better for it to simply reflect a dynamic state that updates with new fee estimations. Also, it’s quite rare for a user to hit the exact maximum number on their own. Furthermore, using estimates to display this value could mislead users in such a critical part of our process. In my opinion, we should treat this as a separate task with additional product considerations

Reply:
@JoseRFelix  @jonator I feel this unsatisfactory. That is I am not sure it is that rare for a user to hit the exact maximum number on their own (I personally keep doing that when I am trying things out) nor I am sure that it is not part of the app to provide assurances for. Additionally leaving a complex state without tests feels very risky for me.

However, I don't think it's a good idea to keep this PR waiting because it's pretty good. So I'm resolving this, although the concern is not.

Reply:
I appreciate your input on this. I see you’ve already created a ticket for it in Linear. Let’s keep the conversation going and aim to resolve this in a separate PR using that ticket. Thank you 🙌